server/routes/receive/dashboardPresenter.ts

```diff
@@ -0,0 +1,38 @@
+import { SentReferral, ServiceCategory } from '../../services/interventionsService'
+import CalendarDay from '../../utils/calendarDay'
+import ReferralDataPresenterUtils from '../referrals/referralDataPresenterUtils'
+import utils from '../../utils/utils'
+
+export default class DashboardPresenter {
+  constructor(private readonly referrals: SentReferral[], private readonly serviceCategories: ServiceCategory[]) {}
+
+  readonly tableHeadings = ['Date received', 'Referral', 'Service user', 'Intervention type']
+
+  readonly tableRows: { text: string; sortValue: string | null }[][] = this.referrals.map(referral => {
+    const { serviceCategoryId } = referral.referral
+    const serviceCategory = this.serviceCategories.find(aServiceCategory => aServiceCategory.id === serviceCategoryId)
+    if (serviceCategory === undefined) {
+      throw new Error(`Expected serviceCategories to contain service category with ID ${serviceCategoryId}`)
+    }
+
+    const day = CalendarDay.parseIso8601(referral.sentAt)
+    let formattedDay = null
```

Comment:
Not a blocker, but this might be a bit cleaner as: `const formattedDay = day ? ReferralDataPresenterUtils.govukShortFormattedDate(day) : ''`

Reply:
Yeah, I think that's nicer, will change it.